src/Altinn.Platform/Altinn.Platform.Events/Tests/TestingRepositories/EventsRepositoryTest.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Threading.Tasks;
+using Altinn.Platform.Events.Models;
+using Altinn.Platform.Events.Repository;
+using Altinn.Platform.Events.Tests.Mocks;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Altinn.Platform.Events.Tests.TestingRepositories
+{
+    /// <summary>
+    /// A collection of tests related to <see cref="EventsRepository"/>.
+    /// </summary>
+    public class EventsRepositoryTest
+    {
+        private readonly Mock<ILogger<EventsRepository>> _loggerMock = new Mock<ILogger<EventsRepository>>();
+
+        /// <summary>
+        /// Scenario:
+        ///   Store a cloud event in Cosmos DB.
+        /// Expected result:
+        ///   Returns the id of the newly created document.
+        /// Success criteria:
+        ///   The response is a non-empty string.
+        /// </summary>
+        [Fact]
+        public async Task Create_EventSuccessfullyStored_IdReturned()
+        {
+            // Arrange
+            EventsRepository repository = new EventsRepository(new EventsCosmosServiceMock(), _loggerMock.Object);
+
+            CloudEvent cloudEvent = new CloudEvent
+            {
+                Specversion = "1.x - wip",
+                Source = new Uri("http://www.altinn.no"),
+                Subject = "party/234234422"
+            };
+
+            // Act
+            string actual = await repository.Create(cloudEvent);
+
+            // Assert
+            Assert.NotEmpty(actual);
+        }
+
+        /// <summary>
+        /// Scenario:
+        ///   Store a cloud event in Cosmos DB, but an unexpected error occurs when storing the trigger function.
+        /// Expected result:
+        ///   An exception thrown.
+        /// Success criteria:
+        ///   An exception is thronw.
+        /// </summary>
+        [Fact]
+        public async Task Create_StoreTriggerException_CritialEventLogged()
+        {
+            // Arrange
+            EventsRepository repository = new EventsRepository(new EventsCosmosServiceMockFails(false, true), _loggerMock.Object);
+
+            try
+            {
+                // Act
+                await repository.Create(new CloudEvent());
```

Comment:
We can verify that a test target throw the expected exception, but in the case of repository.Create method there are no logic for throwing exceptions. It is letting exceptions produced by its dependency pass through.

Reply:
The thought here, and for the method above is to ensure that if cosmos throws an exception this is not hidden by the repositories class but forwarded to the controller

Reply:
I like what you're thinking. This is a special case where the class that depend on of this class expects exceptions and has exception handling, but we have now moved that down one dependency layer. "Clearly we need a way to ensure the status quo."

I would also want some way of verifying the flow here, but it is going to be e very long road ahead if we are to start testing for absence of code. Maybe there is another way to test this, or maybe this is a sign of a code smell or some flaw in our implementation pattern. I'm not sure.

Reply:
removed test